src/RestWrite.js

```diff
@@ -399,8 +399,39 @@ RestWrite.prototype.transformUser = function() {
             });
         }
       }
+      // check whether password is repeating from specified history
```

Comment:
Yes. I too felt its become somewhat big and kind of difficult to comprehend. Too many things going on there, more so after I added passwordPolicy. I will try and refactor it.